feature: Support Azure Managed identities on Azure worker nodes with Fusion (#6118)
nextflow-io/nextflow: docs/reference/config.md

`azure.batch.pools.<name>.lowPriority`
: Enable the use of low-priority VMs (default: `false`).

`azure.batch.pools.<name>.managedIdentityId`
: :::{versionadded} 25.01.0-edge
  :::

> : Specify the pool has a managed identity attached. This is only compatible with Fusion. This is passed to Fusion as the environment variable `FUSION_AZ_MSI_CLIENT_ID`.

Reviewer: ```suggestion
: Specify the pool with an attached managed identity. Defaults to environment variable `FUSION_AZ_MSI_CLIENT_ID`.

: :::{note}
  This feature is only compatible with Fusion.
  :::
```
The suggestions here are to bring it in line with the wording and formatting on the rest of the page. Please check if the meaning is retained. I didn't understand the first sentence or the relationship with the env variable.
